src/MicroHs/List.hs

```diff
@@ -12,11 +12,30 @@ import Data.List
 elemBy :: (a -> a -> Bool) -> a -> [a] -> Bool
 elemBy eq a = any (eq a)
 
--- A simple "quicksort" for now.
+-- A simple merge sort for now.
 sortLE :: forall a . (a -> a -> Bool) -> [a] -> [a]
-sortLE _  [] = []
-sortLE le (x:xs) = sortLE le lt ++ (x : sortLE le ge)
-  where (ge, lt) = partition (le x) xs
+sortLE le = mergeAll . splatter
+  where
+    splatter [] = []
+    splatter [a] = [[a]]
+    splatter (a1 : a2 : as)
+      | a1 `le` a2 = [a1, a2] : splatter as
+      | otherwise  = [a2, a1] : splatter as
```

Comment:
In GHC, using this sort of paired splatter significantly improves performance, by avoiding allocation churn. I don't know if works the same way in MicroHs; if not, we can just use `splatter = map (: [])`.